js/classes/zone.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * A Zone is a small portion of the Game World. 
+ * The Game Engine will load and deal with one Zone at a time.
+ * A Dimension (once an actual class, now simply an in-game concept) will contain several Zones.
+ * Zones should not be created willy-nilly - the only way you should actually construct a Zone is from the static getZones() method!
+ * 
+ * A Zone will define the minimum (upper left) and maximum (bottom right) point which is going to be visible to the Camera -
+ * anything outside this range will never be visible.
+ * A Zone will also contain a load method: this ensures that all assets are downloaded, all entities are placed (INCLUDING CHAD), etc.
+ * 
+ * @author Devin Peevy
+ */
+class Zone {
+
+    /**
+     * This constructor is PRIVATE should NOT be used outside the Zone class. To load a Zone, use the Zone.getZones() static method.
+     * This constructor can be used INSIDE THAT METHOD to dictate the min/max point and loadFunction.
+     * @param {Vector} minBlock The FIRST (top left) block which should be visible in this Zone. (NOT PIXEL COORDINATES)
+     * @param {Vector} maxBlock The LAST (bottom right) block which should be visible in this Zone. (NOT PIXEL COORDINATES)
+     * @param {function} loadFunction A function which is going to download all necessary assets and add all necessary entities into the zone.
+     */
+    constructor(minBlock, maxBlock, loadFunction) {
```

Comment:
Wow, I didn't know you could pass a function call as a parameter like this! That is so neat.

Reply:
JS is sick sometimes B)